internal/datastore/postgres/postgres.go

```diff
@@ -106,11 +106,22 @@ func NewPostgresDatastore(
 
 	configurePool(config, pgxConfig)
 
-	dbpool, err := pgxpool.ConnectConfig(context.Background(), pgxConfig)
+	initializationContext, cancelInit := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancelInit()
+
+	dbpool, err := pgxpool.ConnectConfig(initializationContext, pgxConfig)
 	if err != nil {
 		return nil, fmt.Errorf(errUnableToInstantiate, err)
 	}
 
+	// Verify that the server supports commit timestamps
+	var trackTSOn string
+	if err := dbpool.
+		QueryRow(initializationContext, "SHOW track_commit_timestamp;").
+		Scan(&trackTSOn); err != nil {
+		return nil, fmt.Errorf(errUnableToInstantiate, err)
```

Comment:
I think we need a more descriptive error here if this feature is not enabled, ideally one indicating the exact documentation on how to enable it

Reply:
Do we want to skip this check if watch is disabled?

Reply:
The datastore has no idea if the watch server is enabled.

Reply:
Maybe we should thread that in then?